File: internal/handler/response.go
package handler

import (
	"net/http"

	"github.com/gin-gonic/gin"
	"things-expired/pkg/errors"
)

// Response 统一响应结构
type Response struct {
	Code    int         `json:"code"`
	Message string      `json:"message"`
	Data    interface{} `json:"data"`
}

// Success 返回成功响应
func Success(c *gin.Context, data interface{}) {
	c.JSON(http.StatusOK, Response{
		Code:    0,
		Message: "success",
		Data:    data,
	})
}

// Fail 返回错误响应
func Fail(c *gin.Context, err error) {
	if appErr, ok := err.(*errors.AppError); ok {
		c.JSON(http.StatusOK, Response{
			Code:    appErr.Code,
			Message: appErr.Message,
			Data:    nil,
		})
		return
	}

	// 未知错误
	c.JSON(http.StatusOK, Response{
		Code:    errors.CodeInternalError,
		Message: "内部错误",
		Data:    nil,
	})
}

// FailWithCode 返回带错误码的响应
func FailWithCode(c *gin.Context, code int, message string) {
	c.JSON(http.StatusOK, Response{
		Code:    code,
		Message: message,
		Data:    nil,
	})
}

File: src/utils/response.ts
import type { Context } from "hono";
import { CodeSuccess, CodeInternalError } from "../errors/code";
import { AppError } from "../errors";

interface ResponseBody {
  code: number;
  message: string;
  data: unknown;
}

export function success(c: Context, data: unknown): Response {
  return c.json({
    code: CodeSuccess,
    message: "success",
    data,
  } satisfies ResponseBody);
}

export function failWithCode(
  c: Context,
  code: number,
  message: string,
): Response {
  return c.json({
    code,
    message,
    data: null,
  } satisfies ResponseBody);
}

export function fail(c: Context, error: unknown): Response {
  if (error instanceof AppError) {
    return c.json({
      code: error.code,
      message: error.message,
      data: null,
    } satisfies ResponseBody);
  }

  // 未知错误
  console.error("Unhandled error:", error);
  return c.json({
    code: CodeInternalError,
    message: "内部错误",
    data: null,
  } satisfies ResponseBody);
}
